x-pack/plugins/triggers_actions_ui/public/application/sections/common/components/rule_quick_edit_buttons.tsx

```diff
@@ -20,225 +20,288 @@ import { useKibana } from '../../../../common/lib/kibana';
           {
-            defaultMessage: 'Failed to update API keys for rule(s)',
+            defaultMessage: 'Failed to snooze/unsnooze rule(s)',
           }
         ),
       });
     } finally {
-      setIsUpdatingRuleAPIKeys(false);
+      setIsSchedulingRules(false);
       onActionPerformed();
     }
   }
 
   return (
     <EuiFlexGroup alignItems="baseline" direction="column" gutterSize="none">
-      {!allRulesMuted && (
-        <EuiFlexItem>
-          <EuiButtonEmpty
-            onClick={onmMuteAllClick}
-            isLoading={isMutingRules}
-            isDisabled={isPerformingAction || hasDisabledByLicenseRuleTypes}
-            data-test-subj="muteAll"
-          >
-            <FormattedMessage
-              id="xpack.triggersActionsUI.sections.rulesList.bulkActionPopover.muteAllTitle"
-              defaultMessage="Snooze indefinitely"
-            />
-          </EuiButtonEmpty>
-        </EuiFlexItem>
-      )}
-      {allRulesMuted && (
-        <EuiFlexItem>
-          <EuiButtonEmpty
-            onClick={onUnmuteAllClick}
-            isLoading={isUnmutingRules}
-            isDisabled={isPerformingAction || hasDisabledByLicenseRuleTypes}
-            data-test-subj="unmuteAll"
-          >
-            <FormattedMessage
-              id="xpack.triggersActionsUI.sections.rulesList.bulkActionPopover.unmuteAllTitle"
-              defaultMessage="Cancel snooze"
-            />
-          </EuiButtonEmpty>
-        </EuiFlexItem>
-      )}
-      {allRulesDisabled && (
-        <EuiFlexItem>
-          <EuiButtonEmpty
-            onClick={onEnableAllClick}
-            isLoading={isEnablingRules}
-            isDisabled={isPerformingAction || hasDisabledByLicenseRuleTypes}
-            data-test-subj="enableAll"
-          >
-            <FormattedMessage
-              id="xpack.triggersActionsUI.sections.rulesList.bulkActionPopover.enableAllTitle"
-              defaultMessage="Enable"
-            />
-          </EuiButtonEmpty>
-        </EuiFlexItem>
-      )}
-      {!allRulesDisabled && (
-        <EuiFlexItem>
-          <EuiButtonEmpty
-            onClick={onDisableAllClick}
-            isLoading={isDisablingRules}
-            isDisabled={isPerformingAction || hasDisabledByLicenseRuleTypes}
-            data-test-subj="disableAll"
-          >
-            <FormattedMessage
-              id="xpack.triggersActionsUI.sections.rulesList.bulkActionPopover.disableAllTitle"
-              defaultMessage="Disable"
-            />
-          </EuiButtonEmpty>
-        </EuiFlexItem>
-      )}
+      <EuiFlexItem>
+        <EuiButtonEmpty
+          onClick={onSnoozeAllClick}
+          isLoading={isSnoozingRules}
+          isDisabled={isPerformingAction || hasDisabledByLicenseRuleTypes}
+          data-test-subj="bulkSnooze"
+        >
+          <FormattedMessage
+            id="xpack.triggersActionsUI.sections.rulesList.bulkActionPopover.snoozeAllTitle"
+            defaultMessage="Snooze"
```

Comment:
Perhaps this has already been discussed and dismissed, but from a text point of view, it would be simplest if this was split into four different menu options:

- Snooze now
- Unsnooze now
- Schedule snooze (or Add snooze schedules)
- Unschedule snooze (or Remove snooze schedules)

Reply:
I think we should call it `snooze immediately` like we do in the pop up and then user will realize that they can snooze immediately for just 4 hrs if they want.

Reply:
> I think we should call it `snooze immediately` like we do in the pop up and then user will realize that they can snooze immediately for just 4 hrs if they want.

I am completely wrong here, @lcawl your idea is awesome, let's do that @JiaweiWu!